### PR TITLE
[#143] Should validate on open or project initialization, should not …

### PR DESCRIPTION
**Pull request by ghillairet**
…require edit

### CODE REVIEW

File: com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java [hunk 1]
```diff
@@ -181,6 +183,8 @@ protected void doSetInput(IEditorInput input) throws CoreException {
 			IDocument document = getDocumentProvider().getDocument(getEditorInput());
 			if (document != null) {
 				document.addDocumentListener(changeListener);
+				// validate content before editor opens
+				validate();
```
Comment (tfesenko) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java]: Nice

**Comment by tfesenko** [comment]
Malformed YAML is not detected, probably because `yamlError` is not initialized on `com.reprezen.swagedit.editor.SwaggerEditor.doSetInput(IEditorInput)` yet as it's initialized on `com.reprezen.swagedit.editor.SwaggerDocument.onChange()`

To create a malformed YAML, I took a regular Swagger model and shifted one of the fields.

**Comment by ghillairet** [comment]
The last commit fixes that problem, please have a look.

File: com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java [hunk 2]
```diff
@@ -282,16 +282,16 @@ public void updateFoldingStructure(List<Position> positions) {
...
 
 	@Override
 	public void doSaveAs() {
 		super.doSaveAs();
-		validate();
+		validate(false);
 	}
 
-	protected void validate() {
+	protected void validate(boolean onOpen) {
```
Comment (tfesenko) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java:298]: `validate(true)` is used only once - on `doSetInput()`, in all other cases it's `validate(false)`. You could have left the default implementation like this:

``` java
protected void validate() {
  validate(false);
}
protected void validate(boolean onOpen) { <...>}
```

Comment (ghillairet) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java:298]: True, I updated the PR.

**Comment by tfesenko** [comment]
Looks good!

**Pull request merged**